src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/ApplicationAssembliesProvider.cs

```diff
@@ -0,0 +1,256 @@
+        /// </para>
+        /// </para>
+        /// </summary>
+        public IEnumerable<Assembly> ResolveAssemblies(Assembly entryAssembly)
+        {
+            var dependencyContext = LoadDependencyContext(entryAssembly);
+
+            IEnumerable<AssemblyItem> assemblyItems;
+
+            if (dependencyContext == null)
+            {
+                assemblyItems = new[] { GetAssemblyItem(entryAssembly) };
+            }
+            else
+            {
+                assemblyItems = ResolveFromDependencyContext(dependencyContext);
+            }
+
+            assemblyItems = assemblyItems
+                .OrderBy(item => item.Assembly == entryAssembly ? 0 : 1)
+                .ThenBy(item => item.Assembly.FullName, StringComparer.Ordinal);
+
+            foreach (var item in assemblyItems)
+            {
+                yield return item.Assembly;
+
+                foreach (var associatedAssembly in item.RelatedAssemblies.OrderBy(assembly => assembly.FullName, StringComparer.Ordinal))
+                {
+                    yield return associatedAssembly;
+                }
+            }
+        }
+
+        protected virtual DependencyContext LoadDependencyContext(Assembly assembly) => DependencyContext.Load(assembly);
+
+        private List<AssemblyItem> ResolveFromDependencyContext(DependencyContext dependencyContext)
+        {
+            var assemblyItems = new List<AssemblyItem>();
+            var associatedAssemblies = new HashSet<Assembly>();
+
+            var candidateAssemblies = GetCandidateLibraries(dependencyContext)
+                .SelectMany(library => GetLibraryAssemblies(dependencyContext, library));
+
+            foreach (var assembly in candidateAssemblies)
+            {
+                var assemblyItem = GetAssemblyItem(assembly);
+                assemblyItems.Add(assemblyItem);
+
+                foreach (var associatedAssembly in assemblyItem.RelatedAssemblies)
+                {
+                    associatedAssemblies.Add(associatedAssembly);
+                }
+            }
+
+            // Remove any top level assemblies that appear as an associated assembly.
+            assemblyItems.RemoveAll(item => associatedAssemblies.Contains(item.Assembly));
+
+            return assemblyItems;
+        }
+
+        protected virtual IEnumerable<Assembly> GetLibraryAssemblies(DependencyContext dependencyContext, RuntimeLibrary runtimeLibrary)
+        {
+            foreach (var assemblyName in runtimeLibrary.GetDefaultAssemblyNames(dependencyContext))
+            {
+                var assembly = Assembly.Load(assemblyName);
+                yield return assembly;
+            }
+        }
+
+        protected virtual AssemblyItem GetAssemblyItem(Assembly assembly)
+        {
+            // Do not require related assemblies to be available in the default code path.
```

Comment:
RazorSdk may stamp in a `RelatedAssemblyAttribute` without producing a view. See https://github.com/aspnet/Razor/pull/2154#discussion_r174235581

Reply:
Is this temporary?

Reply:
No, there isn't a way to address this. You can opt in to RazorCompileOnPublish but not build. When `Build` is called, we have to stamp the attribute, but it's not trivial to determine if you're compiling for `Build` or for `Publish`.